consumer_test.go

```diff
@@ -33,6 +33,14 @@ var _ = Describe("Consumer", func() {
 		}()
 	}
 
+	It("should fail for invalid group id", func() {
+		_, err := newConsumer("")
+		Expect(err).To(HaveOccurred())
+
+		_, err = newConsumer("foo:bar")
```

Comment:
That is confusing. I ran into the `client.id` issue in sarama, but only noticed it because metrics subsystem was choking in the Kafka logs, not because it explicitly complained about an invalid character. So I assumed that since those two properties were being checked in the Java source, it would be enforced.. I just tested a group id with the scripts provided with Kafka, and it appears forward slashes are valid too!

I do have an issue open in the Kafka bug tracker (https://issues.apache.org/jira/browse/KAFKA-3417) to try and get to the bottom of this. I'd prefer to wait until we get some clarity on what is valid or not.